SPECS/golang/golang-1.17.spec

```diff
@@ -1,236 +0,0 @@
-%global goroot          %{_libdir}/golang
```

Comment:
We might want to rebase because of:
- #7508
- #7511

Also, with the current state, I was able to build locally :)

Reply:
> Also, with the current state, I was able to build locally :)

Good to know, we couldn't get the buddy build to pass but I'm not sure it was our spec file at fault.

@dagood can you rebase?

Reply:
Rebased onto the latest build, https://dev.azure.com/mariner-org/mariner/_build/results?buildId=493968, I'm fairly sure.

Kicked off this buddy build with the build number: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=495337
And this one with `latest`: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=495338&view=results